airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io.airbyte.integrations.source.mongodb/MongoDbSource.java

```diff
@@ -135,7 +135,26 @@ protected JsonSchemaPrimitive getType(final BsonType fieldType) {
     return tableInfos;
   }
 
-  @Override
+    private Set<String> getAuthorizedCollections(MongoDatabase database) {
+      /* db.runCommand ({listCollections: 1.0, authorizedCollections: true, nameOnly: true })
+       the command returns only those collections for which the user has privileges.
+       For example, if a user has find action on specific collections, the command returns only those collections;
+       or, if a user has find or any other action, on the database resource, the command lists all collections in the database.
+      */
+        Document document = database.getDatabase().runCommand(new Document("listCollections", 1)
+                .append("authorizedCollections", true).append("nameOnly", true));
+
+        return document.toBsonDocument()
+                .get("cursor").asDocument()
+                .getArray("firstBatch")
+                        .stream()
+                .filter(bsonValue -> bsonValue.asDocument().getString("type").getValue().equals("collection"))
```

Comment:
Nitpick .The `listCollections` command can take in a `filter` parameter to filter on any field in the returned document directly.

Reference: https://docs.mongodb.com/manual/reference/command/listCollections/#definition

I think if you add the filter parameter, you won't need to do the filter here:
```java
Document document = database.getDatabase().runCommand(new Document("listCollections", 1)
  .append("authorizedCollections", true)
  .append("nameOnly", true))
  .append("filter", "{ 'type': 'collection' }")
```

I am not 100% sure this is the correct syntax though.